.cursor/rules/component-selection-guide.mdc

```diff
@@ -0,0 +1,125 @@
+---
+alwaysApply: true
+---
+
+# Component Selection: Use Semantic Components Over Generic Styling
+
+## Principle
+
+Prefer semantic components from the design system over custom CSS implementations. This ensures consistency, reduces maintenance, and improves accessibility.
+
+## Guidelines
+
+### Do This ✅
+
+```typescript
+// Good: Use semantic Heading component
+import { Heading } from '@/components/Common'
+
+const PayrollTitle = () => (
+  <Heading level={2} size="lg">
+    Payroll Overview
+  </Heading>
+)
```

Comment:
Looks like this example was hallucinated, we should update to use actual heading props with `as`